lib/protein/adapters/amqp_adapter/server.ex

```diff
@@ -91,8 +90,22 @@ defmodule Protein.AMQPAdapter.Server do
   defp init_conn_chan_queue(url, queue) do
     case Connection.open(url) do
       {:ok, conn} ->
-        {:ok, chan} = Channel.open(conn)
-        Queue.declare(chan, queue)
+        chan =
+          try do
+            {:ok, chan} = Channel.open(conn)
+            {:ok, _} = Queue.declare(chan, queue, durable: true)
+            Logger.info("Declared queue #{queue} as durable")
+            chan
+          catch
+            :exit, reason = {{:shutdown, {:server_initiated_close, 406, _message}}, _} ->
```

Comment:
I'm wondering if this isn't too broad - you don't seem to only catch a durability related message. Perhaps you could either pattern-match or test later if message is relevant to durability?

Reply:
`406` code stands for `PRECONDITION_FAILED` which is directly related to calling `Queue.declare` with mismatched (in relation to the server settings) parameters.

Reply:
Fair enough :+1: